ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py

```diff
@@ -125,3 +125,16 @@ def rfftfreq(n, d=1.0):
     N = n // 2 + 1
     results = ivy.arange(0, N, dtype=int)
     return results * val
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.24.3 and above": ("float16",)}, "numpy")
+def rfft2(a, s=None, axes=(-2, -1), norm=None):
+    a = ivy.array(a, dtype=ivy.float64)
+    if norm is None:
+        norm = "backward"
+    # if s is None:
+    #     s = a.shape
+    # else:
+    s = list(s)
```

Comment:
What happens when `s` is None? The `s = list(s)` line will fail. Can you please add the logic for handling that case?